### PR TITLE
Multi contracts support

### PR DESCRIPTION
**Pull request by shunsukew**
Multiple contracts support.
You can scaffold and manage several contract templates under one single swanky project.

### CODE REVIEW

**Comment by shunsukew** [comment]
@codespool 
Please let me know if you have a good idea for naming this command. I use `scaffold` for now.
I think it is also an option to put several commands related to contracts together as subcommands of `swanky contract`.

**Comment by codespool** [comment]
> @codespool Please let me know if you have a good idea for naming this command. I use `scaffold` for now. I think it is also an option to put several commands related to contracts together as subcommands of `swanky contract`. (e.g. `swanky contract new`, `swanky contract call`, `swanky contract deploy`)

I agree, the `swanky contract` command is the way to go.
So `deploy`, `compile` and `call` should all be subcommands (and `new`)

contract name should be the required argument for all, so for example
`swanky contract call flipper`, `swanky contract deploy my_psp22`

**Comment by shunsukew** [comment]
> > @codespool Please let me know if you have a good idea for naming this command. I use `scaffold` for now. I think it is also an option to put several commands related to contracts together as subcommands of `swanky contract`. (e.g. `swanky contract new`, `swanky contract call`, `swanky contract deploy`)
> 
> I agree, the `swanky contract` command is the way to go. So `deploy`, `compile` and `call` should all be subcommands (and `new`)
> 
> contract name should be the required argument for all, so for example `swanky contract call flipper`, `swanky contract deploy my_psp22`

Updated!

**Comment by shunsukew** [comment]
>

**Comment by shunsukew** [comment]
> > > @codespool Please let me know if you have a good idea for naming this command. I use `scaffold` for now. I think it is also an option to put several commands related to contracts together as subcommands of `swanky contract`. (e.g. `swanky contract new`, `swanky contract call`, `swanky contract deploy`)
> > 
> > 
> > I agree, the `swanky contract` command is the way to go. So `deploy`, `compile` and `call` should all be subcommands (and `new`)
> > contract name should be the required argument for all, so for example `swanky contract call flipper`, `swanky contract deploy my_psp22`
> 
> Updated!

As for `swanky contract call` command,
1. 1 contract can have multiple instances with different addresses.
2. personally think swanky-cli should be able to call any contracts which exist on a target chain.

So, rather than taking CONTRACT_NAME as required args, CONTRACT_ADDRESS. this should be also required arg instead of a flag.

**Comment by codespool** [comment]
The `call` command needs a full rewrite, to include what you mentioned, as well as use polkadot.js API, but it didn't make sense to do it until we had multiple contracts and accounts support.
I'll create a separate ticket for it.

File: src/commands/contract/new.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  ]
+
+  async run(): Promise<void> {
+    await ensureSwankyProject();
+
+    const projectPath = path.resolve();
+    const { args, flags } = await this.parse(NewContract);
+
+    if (fs.existsSync(path.join(projectPath, "contracts", args.contractName))) {
```
Comment (codespool) [inline src/commands/contract/new.ts:41]: better to use `pathExists` provided by fs-extra:
https://github.com/jprichardson/node-fs-extra/blob/master/docs/pathExists.md
(or sync version)

File: src/lib/tasks.ts [hunk 2]
```diff
@@ -42,6 +42,14 @@ export async function copyTemplateFiles(
   await copy(contractTemplatePath, path.resolve(projectPath, "contracts", contractName));
```
Comment (codespool) [inline src/lib/tasks.ts:42]: maybe call `copyContractTemplateFiles` here as well

**Review by codespool — CHANGES_REQUESTED** [review]
Couple of minor changes, but great job overall! 👍

File: src/commands/contract/compile.ts [hunk 3]
```diff
@@ -4,7 +4,8 @@ import path = require("node:path");
 import { readdirSync } from "node:fs";
 import { ensureSwankyProject } from "../../lib/command-utils";
 import { Spinner } from "../../lib/spinner";
-export class Compile extends Command {
+import { args } from "@oclif/core/lib/parser";
```
[the remark this answers is not in the document]
Comment (shunsukew) [reply src/commands/contract/compile.ts:7]: sorry it's no need at all

**Review by codespool — APPROVED** [review]
LGTM 👍

**Pull request merged**